test/GovernorCountingFractional.t.sol

```diff
@@ -599,4 +654,273 @@ contract GovernorCountingFractionalTest is Test {
+      "My 1st vote",
+      abi.encodePacked(_firstVote.forVotes, _firstVote.againstVotes, _firstVote.abstainVotes)
+    );
+
+    (uint256 _actualAgainstVotes, uint256 _actualForVotes, uint256 _actualAbstainVotes) =
+      governor.proposalVotes(_proposalId);
+    assertEq(_firstVote.forVotes, _actualForVotes);
+    assertEq(_firstVote.againstVotes, _actualAgainstVotes);
+    assertEq(_firstVote.abstainVotes, _actualAbstainVotes);
+
+    // If the entire weight was cast; further votes are not possible.
+    if (_voter.weight == _actualForVotes + _actualAgainstVotes + _actualAbstainVotes) return;
+
+    // Now cast votes again.
+    VoteData memory _secondVote;
+    _secondVote.forVotes =
+      uint128(_voter.weight.mulWadDown(_voteSplit.percentFor).mulWadDown(_votePercentage2));
+    _secondVote.againstVotes =
+      uint128(_voter.weight.mulWadDown(_voteSplit.percentAgainst).mulWadDown(_votePercentage2));
+    _secondVote.abstainVotes =
+      uint128(_voter.weight.mulWadDown(_voteSplit.percentAbstain).mulWadDown(_votePercentage2));
+
+    vm.prank(_voter.addr);
+    governor.castVoteWithReasonAndParams(
+      _proposalId,
+      _voter.support,
+      "My 2nd vote",
+      abi.encodePacked(_secondVote.forVotes, _secondVote.againstVotes, _secondVote.abstainVotes)
+    );
+
+    (_actualAgainstVotes, _actualForVotes, _actualAbstainVotes) =
+      governor.proposalVotes(_proposalId);
+    assertEq(_firstVote.forVotes + _secondVote.forVotes, _actualForVotes);
+    assertEq(_firstVote.againstVotes + _secondVote.againstVotes, _actualAgainstVotes);
+    assertEq(_firstVote.abstainVotes + _secondVote.abstainVotes, _actualAbstainVotes);
+
+    // If the entire weight was cast; further votes are not possible.
+    if (_voter.weight == _actualForVotes + _actualAgainstVotes + _actualAbstainVotes) return;
+
+    // One more time!
+    VoteData memory _thirdVote;
+    _thirdVote.forVotes =
+      uint128(_voter.weight.mulWadDown(_voteSplit.percentFor).mulWadDown(_votePercentage3));
+    _thirdVote.againstVotes =
+      uint128(_voter.weight.mulWadDown(_voteSplit.percentAgainst).mulWadDown(_votePercentage3));
+    _thirdVote.abstainVotes =
+      uint128(_voter.weight.mulWadDown(_voteSplit.percentAbstain).mulWadDown(_votePercentage3));
+
+    vm.prank(_voter.addr);
+    governor.castVoteWithReasonAndParams(
+      _proposalId,
+      _voter.support,
+      "My 3rd vote",
+      abi.encodePacked(_thirdVote.forVotes, _thirdVote.againstVotes, _thirdVote.abstainVotes)
+    );
+
+    (_actualAgainstVotes, _actualForVotes, _actualAbstainVotes) =
+      governor.proposalVotes(_proposalId);
+    assertEq(_firstVote.forVotes + _secondVote.forVotes + _thirdVote.forVotes, _actualForVotes);
+    assertEq(
+      _firstVote.againstVotes + _secondVote.againstVotes + _thirdVote.againstVotes,
+      _actualAgainstVotes
+    );
+    assertEq(
+      _firstVote.abstainVotes + _secondVote.abstainVotes + _thirdVote.abstainVotes,
+      _actualAbstainVotes
+    );
+  }
```

Comment:
I feel like this should be enough. Given [5000 fuzz runs](https://github.com/ScopeLift/flexible-voting/blob/debfa653e2e18d2421c9aa405db228baee58f8a1/foundry.toml#L11) the odds seem pretty good that we end up with at least one in which all of the weight is cast.

To make sure, though, I'll add a couple non-fuzz tests that specify the percentages so that we certainly will run tests that exercise both full weight and partial weight

Reply:
Done in 575f596